### PR TITLE
Day 2

### PR DESCRIPTION
**Pull request by daria-sieroshtan**

### CODE REVIEW

File: FlightGraph.php [hunk 1]
```diff
@@ -47,6 +47,16 @@ public function getConnections()
        return $this->connections;
     }
 
+    public function getKeys()
+    {
+        return $this->keys;
+    }
+
+    public function getDestinations($node)
```
Comment (arodiss) [inline FlightGraph.php]: @daria-sieroshtan 
you don't need this as long as you have `ArrayAccess` implemented

File: FlightGraph.php [hunk 2]
```diff
@@ -124,3 +134,57 @@ function($node, $possibleDestinations) {
...
+        return null;
+    }
+    if (!is_int($maxHops) || $maxHops <=0) {
+        echo "Number of hops should be natural number.  \n";
+        return null;
+    } else {
+        $currentRoutes = [];
+        foreach ($graph[$source] as $item){
+            $currentRoutes[] = [$source, $item];
+        }
```
Comment (arodiss) [inline FlightGraph.php]: @daria-sieroshtan 
no need to do this "first hop analysis" manually
Just initialize `$currentRoutes` as `[$source]`

File: FlightGraph.php [hunk 3]
```diff
@@ -124,3 +134,57 @@ function($node, $possibleDestinations) {
...
+        for($i = 1; $i <= $maxHops; ++$i) {
```
Comment (arodiss) [inline FlightGraph.php]: @daria-sieroshtan 
I think you will have one extra hop allowed with this
Please check this and fix if necessary

File: FlightGraph.php (hunk 1 above)
Comment (arodiss) [reply FlightGraph.php]: PS. This interface also provides `offsetExists()` which corresponds to `isset($haystack[$needle])`

Comment (daria-sieroshtan) [reply FlightGraph.php]: I added this because of  "Your graph now needs to expose public method getDestinations($node)" requirement.

Comment (arodiss) [reply FlightGraph.php]: @daria-sieroshtan 
your mentor is an idiot

**Pull request merged**